docs/DeveloperGuide.adoc

```diff
@@ -262,6 +330,239 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+
+Given below is an example usage scenario of the `UniqueTagList#addTag(Tag toAdd)` operation and how the module tagging
+mechanism behaves at each step.
+
+Step 1. The user executes the `addtag MODULE_CODE TAG_NAME`. command to add a tag with the name _TAG_NAME_ to the
+module with the module code _MODULE_CODE_.  A `toAdd` variable of type `Tag` and a boolean value `newTagCreated` is
+maintained during the execution of the command to represent the tag that is to be added and whether or not a new tag
+has been created respectively. The `addtag` command calls `Model#activeSpContainsTag(String tagName)` to check if the
+active study plan contains a tag with the given `tagName`. This method accesses the active study plan in the module
+planner and checks if such a tag exists in its `UniqueTagList`. There are two possible scenarios that are described
+in steps 2 and 3.
+
+Step 2. If such a tag does not exist, a new `UserTag` is created and is assigned to `toAdd`.
+
+image::TagModuleCommandDiagram1.png[]
+
+Step 3. If such a tag exists, `Model#getTagFromActiveSp(String tagName)` is called. The `Tag` in the `UniqueTagList`
+of the active study plan is returned and assigned to `toAdd`. `Tag#isDefault()` is called to check if the returned `Tag`
+is a `DefaultTag`. If so, a `CommandException` is thrown as default tags cannot be added by users.
+
+image::TagModuleCommandDiagram2.png[]
+
+[NOTE]
+We only have to handle the case of adding default tags in this step and not in the previous step as default tags are
+already initialised into the study plan and hence `Model#activeSpContainsTag(String tagName)` will always return true
+if the given name is a default tag name.
+
+Step 4. `Model#addTagToActiveSp(UserTag toAdd, String moduleCode)` is called to add `toAdd` to the module with the
+given module code. This method accesses the module with the given `moduleCode`, which will call `Module#addTag(Tag tag)`
+to add `toAdd` to its `UniqueTagList`. In the case that `toAdd` already exists in the `UniqueTagList`, it will not be
+added, and the method will return `false`. (Step 5)
+
+[NOTE]
+The above scenario should not occur if a new tag had been created as described in Step 2.
+
+Otherwise, the `toAdd` will be added and the method will return `true`. (Step 6)
+
+image::TagModuleCommandDiagram3.png[]
+
+image::TagModuleCommandDiagram4.png[]
+
+Step 5. If the tag had not been added, it would indicate that an essentially identical tag had already been attached to
+the target module. Hence, a `CommandException` will be thrown.
+
+[NOTE]
+`Tag` has a method `Tag#isSameTag` to identify essentially identical tags by comparing the tag names for `UserTag`
+(this is case-insensitive) and the `DefaultTagType` for `DefaultTag`.
+
+Step 6: If the tag had been added, a `CommandResult` with a success message is returned.
+
+The following sequence diagram and activity diagram show how the `addtag` command works:
+
+image::TagModuleCommandSequenceDiagram.png[]
```

Comment:
The complexity and physical size of this diagram is enormous. Perhaps you could separate this into several smaller diagrams that are easier to read, or omit some details?

In particular, the `Logic` and `Model` objects can be removed as their operation has been covered in previous sections. The white boxes can also be separated out as reference frames and put into separate diagrams to improve readability.